lib/matplotlib/tests/test_axes.py

```diff
@@ -3289,6 +3289,144 @@ def test_hist_step_bottom():
     ax.hist(d1, bottom=np.arange(10), histtype="stepfilled")
 
 
+def test_hist_stepfilled_geometry():
+    bins = [0, 1, 2, 3]
+    data = [0, 0, 1, 1, 1, 2]
+    _, _, (polygon, ) = plt.hist(data,
+                                 bins=bins,
+                                 histtype='stepfilled')
+
+    xy = [[0, 0], [0, 2], [1, 2], [1, 3], [2, 3], [2, 1], [3, 1],
+                   [3, 0], [2, 0], [2, 0], [1, 0], [1, 0], [0, 0]]
```

Comment:
```suggestion
          [3, 0], [2, 0], [2, 0], [1, 0], [1, 0], [0, 0]]
```
Indent should start alined with the first element inside the list.